docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,190 @@
+```
+
+By using a placeholder like in the example above, the server is going to try to resolve the `my.db.user` variable from:
+
+* A system property using the same key
+* An environment variable by replacing dots with `_`.
+
+Placeholders can also be used when building a server image using the `build` command so that the value of a configuration option is updated
+with the value set to the placeholder (system property or environment variable) when starting the server.
+
+[NOTE]
+You should not use placeholders for configuration options that can only be set through the `build` command. The reason is that these options
+can not change once they are persisted into the server image. Prefer using placeholders only for the configurations options available to
+the `start` command.
+
+== Configuration profiles
+
+Most of the time the server is deployed to different environments like testing, staging, and production environments. Although they share some
+core configuration options (e.g: database, clustering, etc) some other options might change on a per-environment basis.
+In order to make easier to support multiple configurations, the server allows you to create configuration profiles.
+
+Configuration profiles are basically a set of one or more configuration options grouped together using a unique key.
+
+You can activate a profile by using the `[-pf|--profile]` option as follows:
+
+<@kc.all rootParameters="-pf myprofile"/>
+
+Once a profile is activated, all the configuration options defined for the profile should take effect when configuring or starting the server.
+
+=== Understanding the default profiles
+
+By default, the server defines two main profiles:
+
+* `dev`
+* `prod`
+
+The `dev` profile is activated everytime you run the `start-dev` command. In this profile, some key configuration options are set to make possible to start the
+server for development purposes without the burden of having to define additional settings that are mandatory for production.
+
+The `prod` profile is activated by default whenever you run the `build` or the `start` command. You should use this profile to set any configuration option that
+makes sense when deploying the server in production.
+
+By default, the configurations options for the `prod` profile are commented in the `conf/keycloak.properties`. By looking at these options
+you should have an idea about the main settings that you should consider when running in production.
+
+You should be able to change the configuration options for these built-in profiles as well as create your own.
+
+=== Defining a configuration profile
+
+A profile can be created by adding a set of one or more properties to the `conf/keycloak.properties` file as follows:
+
+```
+%myprofile.db.username=foo
+%myprofile.db.password=foo_secret
+%myprofile.hostname=mykeycloak
+```
+
+Any configuration option associated with a profile should be prefixed by a `%` followed by the profile name and a dot `.` like in the example above.
```

Comment:
"If no profile is given for a specific property, the global property is used instead, so take care what you add to the global namespace" or sth?

Reply:
Good point. Going to add a note.